source/Plugins/ExpressionParser/Swift/SwiftExpressionParser.cpp

```diff
@@ -543,6 +543,13 @@ AddRequiredAliases(Block *block, lldb::StackFrameSP &stack_frame_sp,
   if (!imported_self_type.IsValid())
     return;
 
+  SwiftLanguageRuntime *swift_runtime =
+      stack_frame_sp->GetThread()->GetProcess()->GetSwiftLanguageRuntime();
+  auto *stack_frame = stack_frame_sp.get();
+  imported_self_type =
+      swift_runtime->DoArchetypeBindingForType(*stack_frame,
+                                               imported_self_type);
+
```

Comment:
I think this is reasonable.
Whether the compiler does the archetype binding for us or we do it for him doesn't make a difference in the end. And the difference between `expr` and `expr -d no-runtime` is one of presenting the result, not really the input expression, at least as far as generic types are concerned.

Reply:
@jimingham may have an opinion on this?